application_authentication_handlers_test.go

```diff
@@ -179,3 +181,101 @@ func TestApplicationAuthenticationGetBadRequest(t *testing.T) {
 
 	testutils.BadRequestTest(t, rec)
 }
+
+func TestApplicationAuthenticationCreate(t *testing.T) {
+	if parser.RunningIntegrationTests {
+		t.Skip("Test not supported when using db backend")
+	}
+
+	input := m.ApplicationAuthenticationCreateRequest{
+		ApplicationIDRaw:    7,
+		AuthenticationIDRaw: 7,
+	}
+
+	body, _ := json.Marshal(&input)
+
+	c, rec := request.CreateTestContext(
+		http.MethodPost,
+		"/api/sources/v3.1/application_authentications",
+		bytes.NewBuffer(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+	c.Request().Header.Add("Content-Type", "application/json")
+
+	err := ApplicationAuthenticationCreate(c)
+	if err != nil {
+		t.Error(err)
+	}
+
+	if rec.Code != 201 {
+		t.Errorf("Wrong response code, got %v wanted %v", rec.Code, 201)
+	}
+}
+
+func TestApplicationAuthenticationCreateBadAppId(t *testing.T) {
+	input := m.ApplicationAuthenticationCreateRequest{
+		ApplicationIDRaw:    "abcd",
+		AuthenticationIDRaw: 7,
+	}
+
+	body, _ := json.Marshal(&input)
+
+	c, _ := request.CreateTestContext(
+		http.MethodPost,
+		"/api/sources/v3.1/application_authentications",
+		bytes.NewBuffer(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+	c.Request().Header.Add("Content-Type", "application/json")
+
+	err := ApplicationAuthenticationCreate(c)
+	if err == nil {
+		t.Errorf("Wanted an error but none were returned")
+	}
+}
+
+func TestApplicationAuthenticationCreateBadAuthId(t *testing.T) {
+	input := m.ApplicationAuthenticationCreateRequest{
+		ApplicationIDRaw:    7,
+		AuthenticationIDRaw: "abcd",
+	}
+
+	body, _ := json.Marshal(&input)
+
+	c, _ := request.CreateTestContext(
+		http.MethodPost,
+		"/api/sources/v3.1/application_authentications",
+		bytes.NewBuffer(body),
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+	c.Request().Header.Add("Content-Type", "application/json")
+
+	err := ApplicationAuthenticationCreate(c)
+	if err == nil {
+		t.Errorf("Wanted an error but none were returned")
+	}
+}
+
+func TestApplicationAuthenticationDeleteNotFound(t *testing.T) {
+	c, _ := request.CreateTestContext(
+		http.MethodDelete,
+		"/api/sources/v3.1/application_authentications/1234523452542",
+		nil,
+		map[string]interface{}{
+			"tenantID": int64(1),
+		},
+	)
+	c.SetParamNames("id")
+	c.SetParamValues("1234523452542")
+
+	err := ApplicationAuthenticationDelete(c)
```

Comment:
I am thinking 🤔  in this case we expect 404, so we can check more than that there is some error. For example same logic as is in TestApplicationAuthenticationGetNotFound(). 

```
	notFoundApplicationAuthenticationDelete := ErrorHandlingContext(ApplicationAuthenticationDelete)
	err := notFoundApplicationAuthenticationDelete(c)
	if err != nil {
		t.Error(err)
	}

	testutils.NotFoundTest(t, rec)
```

... and same for new bad request tests ... or  😁. I can update these test after merging in new PR

Reply:
Thanks for catching that, I was completely on auto-pilot and just wrote them the old way :grin:. 

Updated to use the new helpers.